tests/riotboot/tests/01-run.py

```diff
@@ -65,16 +72,26 @@ def testfunc(child):
     child.expect_exact('>')
     child.sendline("curslothdr")
     child.expect(r"Image Version: (?P<app_ver>0x[0-9a-fA-F]{8})\r\n")
-    current_app_ver = int(child.match.group("app_ver"), 16)
-
-    # Flash to both slots and verify basic functions
-    for version in [current_app_ver + 1, current_app_ver + 2]:
-        slotnum = slotnum ^ 1
-        flash_slot(slotnum, version)
-        assert_check_slot(child, slotnum, version)
+    app_ver = int(child.match.group("app_ver"), 16)
 
-    print("TEST PASSED")
+    # Set current slot number and app version
+    global current_app_ver
+    global current_slot_num
+    current_slot_num = slotnum
+    current_app_ver = app_ver
 
 
 if __name__ == "__main__":
-    sys.exit(run(testfunc))
+    res = 1
+    global current_app_ver
+    global current_slot_num
+    # set current slot number and app version
+    res = run(testcurrent)
```

Comment:
You should explain why you have to call the testrunner separately to get the slot number/app version. If possible, I would avoid that.

Reply:
Your concern is with having multiple `run(something)` or just with the specific one pointed out here?

Reply:
Yes multiple run and the global variables. Because a simpler solution seems to just call "testcurrent" directly and change this function to directly return the current slot/app version. That's why I'm asking what is the actual reason for a separate run.

Reply:
Well I'm not very good at this, and so It is indeed dirty, but I'm under the impression that `run()` only returns an exit code, so either `1` or `0` from what I see in code.

What I need to to is `detach` the terminal before flashing because the `cc2538-bsl` script flashes over serial and therefore I can't have the terminal open when flashing.

`run(testfunc)` among others spawns a terminal and the terminal remains attached until `testfunc` exits. And so to be able to flash to `slot0` and `slot1` I need to be able to do the following:

- attach terminal
- get current slot and app version
- detach terminal
- flash to the other slot with newer app version
- attach terminal
- check everything is OK
- detach terminal
- flash to the other slot with newer app version
- attach terminal
- check everything is OK
- test passed

Having multiple `run(testfunc)` allows handling proper attaching/detaching every time

Reply:
Any other suggestion @aabadie?

Reply:
Or is the reasoning still not clear? If its clear I can add a comment in the test script.

Reply:
This is what I was suspecting but this is far from obvious and this why I was initially asking for an explanation (a comment).

Instead of the global keyword, you could maybe wrap things in a class with 2 attributes current_app_ver and current_slot_num. The class would have methods to check the current slot/app version and to perform the test. I think that would be cleaner.

Reply:
Tried to address your comment, let me know what you think